mongodb-aggregate-query-support-test/src/main/java/com/github/krr/mongodb/embeddedmongo/config/LinuxDistributionReader.java

```diff
@@ -0,0 +1,142 @@
+package com.github.krr.mongodb.embeddedmongo.config;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Scanner;
+
+@Slf4j
+public class LinuxDistributionReader {
+  public static String getLinuxVersion() {
+    Map.Entry<String, String> distro = getLinuxDistro();
+    String distribution = distro.getKey();
+    String version = distro.getValue();
+    if (distribution.isEmpty() || version.isEmpty()) {
+      log.error("Error in parsing through os release files to fetch linux distribution and version");
+      return "";
+    }
+    switch (distribution.toLowerCase()) {
+      case "ubuntu":
+        return getUbuntuVersion(version);
+      case "centos":
+      case "rhel":
+        return getRedHatVersion(version);
+      default:
+        // This distribution id isn't supported by mongo
+        log.error("The current linux belongs to {} and isn't supported", distribution);
+        return "";
+    }
+  }
+
+  private static String getRedHatVersion(String version) {
+    int versionId = (int) Double.parseDouble(version);
+    if (versionId <= 6) {
+      return "rhel62";
+    }
+    else if (versionId == 7) {
+      return "rhel70";
+    }
+    else if (versionId == 8) {
+      return "rhel80";
+    }
+    else {
+      // As of this commit, mongo only supports CentOS 6.2+.
+      // Update the ubuntu version link for later versions here
+      log.error("Red Hat {} isn't supported, please contact support", version);
+      return "";
+    }
+  }
+
+  private static String getUbuntuVersion(String version) {
+    int versionId = (int) Double.parseDouble(version);
+    if (versionId <= 16) {
+      return "ubuntu1604";
+    }
+    else if (versionId == 18) {
+      return "ubuntu1804";
+    }
+    else {
+      // As of this commit, mongo only supports Ubuntu-16.04 and Ubuntu-18-04.
+      // Update the ubuntu version link for later versions here
+      log.error("Ubuntu {} isn't supported, please contact support", version);
+      return "";
```

Comment:
Throw a runtime exception here so clients don't have to deal with empty string checks.